### PR TITLE
feat: zrx on bsc and polygon

### LINKED ISSUE #3996: [Demand] ZRX on BSC and Polygon
## Catalogue

- [ ] UI/UX
- [ ] Mask Core Feature
- [x] Mask Plugin Feature (Trader)
- [ ] Security/Cryptography Feature
- [ ] Other
  - /_List it here_/

### PR DESCRIPTION
**Pull request by lelenei**
closes #3996

### CODE REVIEW

File: packages/maskbook/src/plugins/Trader/apis/0x/index.ts [hunk 1]
```diff
@@ -9,7 +11,24 @@ import type {
...
+function resolveZRXLink(networkType: NetworkType) {
+    switch (networkType) {
+        case NetworkType.Binance:
+            return `https://bsc.${ZRX_BASE_URL}`
+        case NetworkType.Ethereum:
+            return `https://${ZRX_BASE_URL}`
+        case NetworkType.Polygon:
+            return `https://polygon.${ZRX_BASE_URL}`
+        case NetworkType.Arbitrum:
+            return `https://${ZRX_BASE_URL}`
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Trader/apis/0x/index.ts:23]: Does it support Arbitrum? If not you can return an empty string.

File: packages/maskbook/src/plugins/Trader/constants/0x.ts [hunk 2]
```diff
@@ -1,2 +1,2 @@
-export const ZRX_BASE_URL = 'https://api.0x.org'
+export const ZRX_BASE_URL = 'api.0x.org'
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Trader/constants/0x.ts:1]: What about just remove this constant?

**Comment by guanbinrui** [comment]
Don't forget to update `packages/maskbook/src/plugins/Trader/apis/trader/index.ts`.

File: packages/maskbook/src/plugins/Trader/apis/0x/index.ts [hunk 3]
```diff
@@ -9,7 +11,24 @@ import type {
     SwapValidationErrorResponse,
 } from '../../types'
 
-export async function swapQuote(request: SwapQuoteRequest) {
+function resolveZRXLink(networkType: NetworkType) {
+    switch (networkType) {
+        case NetworkType.Binance:
+            return `${ZRX_BASE_URL.replace(/(.{8})/, '$1bsc.')}`
```
Comment (septs) [inline packages/maskbook/src/plugins/Trader/apis/0x/index.ts:17]: why not is lookup table?

```ts
const zrxLinks: Record<NetworkType, string> = {
   .....
}
```

File: packages/maskbook/src/plugins/Trader/apis/0x/index.ts [hunk 4]
```diff
@@ -11,24 +10,14 @@ import type {
...
-        case NetworkType.Arbitrum:
-            return ''
-        default:
-            safeUnreachable(networkType)
-            return ''
-    }
+const zrxLink: Record<NetworkType, string> = {
+    [NetworkType.Arbitrum]: '',
+    [NetworkType.Binance]: ZRX_BASE_URL.replace(/(.{8})/, '$1bsc.'),
+    [NetworkType.Polygon]: ZRX_BASE_URL.replace(/(.{8})/, '$1polygon.'),
```
Comment (septs) [inline packages/maskbook/src/plugins/Trader/apis/0x/index.ts:16]: plz do not use regexp replace, add more constants

**Review by septs — DISMISSED** [review]

**Review by Jack-Works — APPROVED** [review]

**Review by Jack-Works — APPROVED** [review]

**Pull request closed without merging**